crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -718,6 +756,174 @@ impl<F: Clone, Ctx: PaymentMethodRetrieve>
         let m_error_message = error_message.clone();
         let m_db = state.clone().store;
 
+        // Validate Blocklist
+        #[cfg(feature = "kms")]
+        let (fingerprint_id, is_pm_blocklisted) = {
+            let merchant_id = payment_data.payment_attempt.merchant_id;
+            let merchant_secret = state
+                .store
+                .find_config_by_key(format!("secret_{}", merchant_id).as_str())
```

Comment:
Create a util function to get the merchant `fingerprint_secret` key instead of using `format!()`